### DISCUSSION TITLE
Upper and lower canvas not aligned

### DISCUSSION THREAD

**Discussion by jjggjg** [fabricjs/fabric.js#9982] [Q&A]
After centering the canvas in his container the upper and lower canvas are not aligned anymore. When selecting a object, the selectionbox doesn't correspond the mouse coordinates. Please advise how to solve this problem. 

upper-canvas: black border
lower-canvas: white border

`<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Fabric.js Voorbeeld</title>
    
    <script src="fabric.min.js"></script>
    <style>
        #canvasContainer {
            background-color: rgb(255, 100, 100);
            width: 1000px;
            height: 1000px;
            border: none;
            position: absolute;
            top: 200px;
            left: 200px;
        }

        #myCanvas {
            position: absolute;
        }

        .lower-canvas {
            border: 1px solid white;
        }

        .upper-canvas {
            border: 1px solid black;
        }

    </style>
</head>

<body>
    <div id="canvasContainer">
        <canvas id="myCanvas"></canvas>
    </div>
    
    <script>
        const container = document.querySelector('#canvasContainer');
        const myCanvas = document.querySelector('#myCanvas');        
        
        var canvas = new fabric.Canvas('myCanvas', {
            width: 500,
            height: 500
        });

        canvas.setBackgroundColor('lightblue', canvas.renderAll.bind(canvas));
        
        let containerWidth = container.clientWidth;
        let containerHeight = container.clientHeight;

        let canvasWidth = canvas.getWidth();
        let canvasHeight = canvas.getHeight();        
        
        myCanvas.style.left = `${(containerWidth - canvasWidth)/2}px`;
        myCanvas.style.top = `${(containerHeight - canvasHeight)/2}px`;

        var rect = new fabric.Rect({
            left: 100,
            top: 100,
            fill: 'white',
            width: 100,
            height: 100
        });
        canvas.add(rect);
        
    </script>
</body>

</html>`

**Comment by asturur** [comment]
Can you put it in a fiddle?
